docs/administrators/policy_cookbook.md

```diff
@@ -0,0 +1,251 @@
+
+This example demonstrates how to share information across multiple PEPs within the Native Rule Engine Plugin.
+
+### How to do it ...
+
+Let's say we have the following rules defined in `/etc/irods/core.re`.
+
+Pay close attention to the use of `temporaryStorage`. It is the bridge between PEPs within the Native Rule Engine Plugin.
+
+```python
+pep_api_data_obj_put_pre(*INSTANCE_NAME, *COMM, *DATA_OBJ_INPUT, *BYTES_BUFFER, *PORTAL_OPR_OUTPUT)
+{
+    # Capture the logical path of the PUT operation.
+    # We can use it later when writing messages to the log file.
+    temporaryStorage."logical_path" = *DATA_OBJ_INPUT.objPath;
+}
+
+pep_api_data_obj_put_post(*INSTANCE_NAME, *COMM, *DATA_OBJ_INPUT, *BYTES_BUFFER, *PORTAL_OPR_OUTPUT)
+{
+    # Fetch the logical path we captured during the call to open.
+    *logical_path = temporaryStorage."logical_path";
+
+    # Write a message to the log file containing the full logical path.
+    writeLine("serverLog", "This string, *logical_path, was captured by the pre-PEP and logged by the post-PEP!");
+}
+```
+
+If you run the following commands:
+```bash
+$ touch foo
+$ iput foo
+```
+
+And inspect the log file at `/var/log/irods/irods.log`, you'd see a log message similar to the following:
+```json
+{
+  "log_category": "legacy",
+  "log_facility": "local0",
+  "log_level": "info",
+  "log_message": "writeLine: inString = This string, /tempZone/home/rods/foo, was captured by the pre-PEP and logged by the post-PEP!\n",
+  "request_api_name": "DATA_OBJ_PUT_AN",
+  "request_api_number": 606,
+  "request_api_version": "d",
+  "request_client_user": "rods",
+  "request_host": "127.0.0.1",
+  "request_proxy_user": "rods",
+  "request_release_version": "rods4.3.0",
+  "server_host": "38832fb94fff",
+  "server_pid": 24234,
+  "server_timestamp": "2022-07-24T16:39:24.348Z",
+  "server_type": "agent"
+}
+```
+
+The prototype implementation of the Logical Quotas Rule Engine Plugin uses this technique. You can find the implementation at [https://github.com/irods/irods_policy_examples/blob/main/irods_policy_logical_quotas.re](https://github.com/irods/irods_policy_examples/blob/main/irods_policy_logical_quotas.re)
+
+Additional documentation can be found at [Using temporaryStorage in the iRODS Rule Language](/system_overview/tips_and_tricks/#using-temporarystorage-in-the-irods-rule-language).
+
+## Simulating User Quotas
+
+This example demonstrates how to simulate user quotas using group quotas.
+
+As of iRODS 4.3.0, support for user quotas has been partially disabled. Please consult `iadmin suq` and the release notes for more information.
+
+### How to do it ...
+
+#### Step 1: Enable Quota Enforcement
+
+First, we have to instruct iRODS to enforce quotas. Open `/etc/irods/core.re` and change the argument passed to `msiSetRescQuotaPolicy()` from `"off"` to `"on"`. The line you modified should look very similar to the one below.
+```python
+acRescQuotaPolicy { msiSetRescQuotaPolicy("on"); }
```

Comment:
I'm not sure what that blurb would say. Do you have words that you feel should be mentioned here?

Reply:
how's this...

> The original RescQuota system was geared to count bytes on storage resources... either per user or group for a single resource, or per user or group across all resources.  Since 4.3.0, user quotas have been removed (user and group quotas could not be enforced simultaneously - see irods/irods#4481)- but group quotas can still be set and enforced across all resources.  In this case, we're interested in setting a quota for a group which contains a single user (equivalent to setting a user quota).

Maybe we should start referring to this quota system as "Physical Quotas"...  wondering out loud if we can pull them out into a separate rule engine plugin...   gah, that sounds like a lot of work.

Reply:
> The original RescQuota system was geared to count bytes on storage resources... either per user or group for a single resource, or per user or group across all resources. Since 4.3.0, user quotas have been removed (user and group quotas could not be enforced simultaneously - see irods/irods#4481)- but group quotas can still be set and enforced across all resources. In this case, we're interested in setting a quota for a group which contains a single user (equivalent to setting a user quota).

My gut reaction is that feels like something that should be linked to rather than placed directly in the cookbook. Although, technical books tend to have little pockets of text like that at various places.

I'm going to add it to see what it's like in the rendered version.

> Maybe we should start referring to this quota system as "Physical Quotas"... wondering out loud if we can pull them out into a separate rule engine plugin... gah, that sounds like a lot of work.

Why "Physical Quotas"? I feel we can externalize them into a REP. Unlike Logical Quotas, this system does not react to data movement in real time, so it may be simpler to implement. Removing the code from the server is probably where the challenge lies.

Reply:
That's fair about living somewhere else and linking... but, not sure where that would be yet.

--

Physical Quotas, rather than Logical Quotas.   Agreed that removing the `iadmin` subcommands would be a hairy process. And providing a(n automatic) transition/upgrade process is not clear either...

Reply:
I mean, I just legitimately didn't know why we had to turn on resc quotas to enable this feature. :)

A single sentence like: "For historical reasons, this policy enables tracking quota usage." Or something like that would be fine by me. Fleshing that out more as @trel did could be valuable too, of course.

Reply:
Sure. I agree that @trel's statement needs to be added to the documentation. I may look into adding that somewhere via this PR.

Is this about protecting the reader from confusion due to the name of the microservice and NREP rule?

Reply:
yes, the Resc in the name.... when we're attempting to set a 'user' quota.

Reply:
I thought so. Take a look at the current write-up and let me know what you think. It is closer to what @alanking proposed.

Reply:
I think it is sufficient, yep

Reply:
Agreed

Reply:
Cool. Marking as resolved.